[비동기 댓글 기능] 러너덕 미션 제출합니다.

Deocksoo opened the pull request:
안녕하세요 브라운, 이번 리뷰이 러너덕입니다.
이번 과제를 급하게 작성하느라 커밋로그를 제대로 정리하지 못했네요ㅜㅜ

[08e30ab] refactor: commentDto의 이름을 RequestCommentDto로 변경합니다

위 커밋에 comment.js 파일이 섞여서 들어가 있습니다. 커밋로그 보시다가 혼란스러우실 수도 있을것 같아 미리 말씀드려요, 혼선을 드려 죄송합니다.
그럼 이번 리뷰도 잘 부탁드립니다 :)

hyperpace committed:
[게시글 생성/조회/수정/삭제] 에이든 미션 제출합니다. (#28)

* doc(sample.md): update sample.md including feature list

* feat(gradle): add thymeleaf library

* feat: update build.gradle

* feat(HelloWorld): update HelloWorld controller

* feat: update ArticleController including create article

* feat(areticle) : Update article controller including find article list

* feat(Article) : Update Article Controller find article by Id

* feat: Remove duplicate html tags

* feat(article) : Update Article Controller including update to select article

* feat(article) : update article controller including delete function

* feat: update script including remove article

* feat: fix bug css

* style: clean up code

* feat(Article): remove setter

* feat(ArticleRepository): rename size to count

* feat(ArticleController): update articleRepository with constructor injection

* feat(MyBlogApplication): rename class Name

* feat(ArticleTest): add ArticleTest

* feat(ArticleRepositoryTest): add ArticleRepositoryTest

* feat(MyBlogApplication): rename class name

* refactor(ArticleControllerTests): abstract methods

* refactor(ArticleControllerTests): abstract methods

hyperpace committed:
[게시판 회원 기능] 에이든 미션 제출합니다. (#72)

* docs : README 작성

* feat : 게시판 생성기능 구현

* docs: readme 수정

* feat : 게시글 및 목록 조회 기능 구현

* feat : sample 게시글 구현

* refactor : html 중복 코드 제거

* refactor : 코드 정리

* feat : 수정 기능 추가

* feat : 게시글 수정 기능 구현

* feat : 게시글 삭제 기능 구현

* refactor : redirect 적용

* feat : Method enum 추가

* test : ArticleRepository test code 추가

* test : ArticleController test code 추가

* refactor : controller method 간소화

* refactor : 설정 위치 변경

* fix : cover url 수정

* fix : css 수정

* fix : index url 수정

* refactor : Mapping 정리, setter injection 적용

* refactor : setter method 제거

- synchronized 적용

* refactor : optional 적용

* test : test code 분리/추가

* refactor : Article id type 변경 (int->long)

- MethodType 제거 -> 기존 enum 사용

* feat : Article에 대한 DTO class 생성

* fix : AtomicLong 적용

* [게시글 생성/조회/수정/삭제] 김고래 미션 제출합니다. (#35)

* docs : README 작성

* feat : 게시판 생성기능 구현

* docs: readme 수정

* feat : 게시글 및 목록 조회 기능 구현

* feat : sample 게시글 구현

* refactor : html 중복 코드 제거

* refactor : 코드 정리

* feat : 수정 기능 추가

* feat : 게시글 수정 기능 구현

* feat : 게시글 삭제 기능 구현

* refactor : redirect 적용

* feat : Method enum 추가

* test : ArticleRepository test code 추가

* test : ArticleController test code 추가

* refactor : controller method 간소화

* refactor : 설정 위치 변경

* fix : cover url 수정

* fix : css 수정

* fix : index url 수정

* refactor : Mapping 정리, setter injection 적용

* refactor : setter method 제거

- synchronized 적용

* refactor : optional 적용

* test : test code 분리/추가

* refactor : Article id type 변경 (int->long)

- MethodType 제거 -> 기존 enum 사용

* feat : Article에 대한 DTO class 생성

* fix : AtomicLong 적용

* feat: init JPA mission

* docs(README.md): Update readme

* feat(build.gradle): add dependency JPA

* feat(Article): update field and validation check

* feat(build.gradle): update mysql dependency

* feat(application.properties): update mysql config

* feat(Article): update constructor

* feat(ArticleRepository): update extends CrudRepository

* fix(index.html): update a href attribute

* feat(ArticleController): create article

* doc(README.md): update feature list

* feat(ArticleRepository): init

* fix(article-edit.html): change form method

* refactor(MainController): delete

* feat(ArticleService): add crud methods

* feat(ArticleController): add article crud methods

* docs(README.md): update feature list

* test(build.gradle): add H2 database for testing

* test(ArticleService): test articleService CRUD

* test(ArticleController): test articleController CRUD

* feat: remove file

* feat(build.gradle): update implements

* style(article): clean up code

* feat(UserDto): add UserDto with field, getter, setter, pattern

* feat(LoginDto): add LoginDto with field, getter, setter, pattern

* feat(application.properties): update setting

* feat(application.properties): update setting

* style(*.html): remove duplicate, clean up

* feat(User): add User with field, constructor

* feat(UserRepository): add UserRepository with option findByEmail

* feat(UserService): add UserService with CRUD, Login

* test(UserService): test UserService with CRUD, Login

* feat(UserController): add UserController with CRUD, login, logout

* Revert "[게시글 생성/조회/수정/삭제] 에이든 미션 제출합니다. (#28)"

This reverts commit b7349040712fe8a76f400945becab6da893c4867.

* style: clean up code

* style: add semicolon

* refactor(Article): Remove of method and add create method to articleDto

* refactor(User): change update method and add create method to userDto

* feat(UserRepository): add existsByEmailAndPassword method and update login method

* feat(application.properties): update user

* refactor(UserController): refactor redirect index(/)

* test(UserController): test UserController with session

* feat(UserService): update exists method

* feat(WrongEmailAndPasswordException): add WrongEmailAndPasswordException and LoginAdvice

* test(ArticleControllerTest): test ArticleController

hyperpace committed:
[댓글 기능] 에이든 미션 제출합니다. (#117)

* docs : README 작성

* feat : 게시판 생성기능 구현

* docs: readme 수정

* feat : 게시글 및 목록 조회 기능 구현

* feat : sample 게시글 구현

* refactor : html 중복 코드 제거

* refactor : 코드 정리

* feat : 수정 기능 추가

* feat : 게시글 수정 기능 구현

* feat : 게시글 삭제 기능 구현

* refactor : redirect 적용

* feat : Method enum 추가

* test : ArticleRepository test code 추가

* test : ArticleController test code 추가

* refactor : controller method 간소화

* refactor : 설정 위치 변경

* fix : cover url 수정

* fix : css 수정

* fix : index url 수정

* refactor : Mapping 정리, setter injection 적용

* refactor : setter method 제거

- synchronized 적용

* refactor : optional 적용

* test : test code 분리/추가

* refactor : Article id type 변경 (int->long)

- MethodType 제거 -> 기존 enum 사용

* feat : Article에 대한 DTO class 생성

* fix : AtomicLong 적용

* [게시글 생성/조회/수정/삭제] 김고래 미션 제출합니다. (#35)

* docs : README 작성

* feat : 게시판 생성기능 구현

* docs: readme 수정

* feat : 게시글 및 목록 조회 기능 구현

* feat : sample 게시글 구현

* refactor : html 중복 코드 제거

* refactor : 코드 정리

* feat : 수정 기능 추가

* feat : 게시글 수정 기능 구현

* feat : 게시글 삭제 기능 구현

* refactor : redirect 적용

* feat : Method enum 추가

* test : ArticleRepository test code 추가

* test : ArticleController test code 추가

* refactor : controller method 간소화

* refactor : 설정 위치 변경

* fix : cover url 수정

* fix : css 수정

* fix : index url 수정

* refactor : Mapping 정리, setter injection 적용

* refactor : setter method 제거

- synchronized 적용

* refactor : optional 적용

* test : test code 분리/추가

* refactor : Article id type 변경 (int->long)

- MethodType 제거 -> 기존 enum 사용

* feat : Article에 대한 DTO class 생성

* fix : AtomicLong 적용

* feat: init JPA mission

* docs(README.md): Update readme

* feat(build.gradle): add dependency JPA

* feat(Article): update field and validation check

* feat(build.gradle): update mysql dependency

* feat(application.properties): update mysql config

* feat(Article): update constructor

* feat(ArticleRepository): update extends CrudRepository

* fix(index.html): update a href attribute

* feat(ArticleController): create article

* doc(README.md): update feature list

* feat(ArticleRepository): init

* fix(article-edit.html): change form method

* refactor(MainController): delete

* feat(ArticleService): add crud methods

* feat(ArticleController): add article crud methods

* docs(README.md): update feature list

* test(build.gradle): add H2 database for testing

* test(ArticleService): test articleService CRUD

* test(ArticleController): test articleController CRUD

* feat: remove file

* feat(build.gradle): update implements

* style(article): clean up code

* feat(UserDto): add UserDto with field, getter, setter, pattern

* feat(LoginDto): add LoginDto with field, getter, setter, pattern

* feat(application.properties): update setting

* feat(application.properties): update setting

* style(*.html): remove duplicate, clean up

* feat(User): add User with field, constructor

* feat(UserRepository): add UserRepository with option findByEmail

* feat(UserService): add UserService with CRUD, Login

* test(UserService): test UserService with CRUD, Login

* feat(UserController): add UserController with CRUD, login, logout

* Revert "[게시글 생성/조회/수정/삭제] 에이든 미션 제출합니다. (#28)"

This reverts commit b7349040712fe8a76f400945becab6da893c4867.

* style: clean up code

* style: add semicolon

* refactor(Article): Remove of method and add create method to articleDto

* refactor(User): change update method and add create method to userDto

* feat(UserRepository): add existsByEmailAndPassword method and update login method

* feat(application.properties): update user

* refactor(UserController): refactor redirect index(/)

* test(UserController): test UserController with session

* feat(UserService): update exists method

* feat(WrongEmailAndPasswordException): add WrongEmailAndPasswordException and LoginAdvice

* test(ArticleControllerTest): test ArticleController

* feat: practice relation mapping

* feat: update database password

* feat: update database option

* feat(Comment): add Comment

* feat(CommentRepository): add CommentRepository

* feat(Article): add Comments and Relationship mapping

* feat(CommentDto): add CommentDto

* feat(CommentService): add CommentService

* feat(CommentDto): add constructor

* feat(CommentServiceTest): create test for save and retrieve

* feat(CommentServiceTest): create test for update comment

* test(CommentServiceTest): create test for delete comment

* feat(Article): add getComments()

* feat(article.html): add Thymeleaf template scaffold for comments

* feat(Comment): add article field

* feat(CommentRepository): add findByArticle method

* feat(CommentService): add findByArticle method

* feat(CommentDto): add article field

* feat(ArticleController): add CommentService injection

* feat(CommentController): add CommentService injection

* feat(article.html): update Thymeleaf template scaffold for comments

* feat: update setting

* feat(Comment): add getter articleId

* feat(CommentService): update delete return type

* feat(CommentController): add update, delete comment method

* feat(article.html): update linked comment-edit

* feat(index.html): update title text

* feat(comment-edit.html): add comment-edit

* test(ArticleControllerTest): update to mockMvc

* test(CommentControllerTest): add commentController test

* refactor(HomeController): refactor HomeController with open index page

* feat(AuthInterceptor): add interceptor including check session

* feat(article): add writer info

* feat(Article): add column property and update toString, eqauls method

* feat(Comment): update toString, equals methods

* feat(User): update column property

* feat(ArticleDto): update field including writer

* style: arrange code style

* doc(README.md): update feature list

* feat(Article): remove setter of Author method

* feat(Comment): add column annotation to contents field

* feat(InterceptorConfig): update path patterns

* feat(AuthInterceptor): update check session

* style(User): arrange code

* style(ArticleDto): remove field of author

* style(CommentDto): remove field of author, article

* refactor: arrange code

* feat: remove annotation @AutoConfigureWebTestClient

* feat(Comment): update access modifier

* feat(InterceptorConfig): update url patterns

* feat(Comment): update code style and fetch setting and join column name

* feat(Article): update code style and fetch setting

* feat(User): update code style and fetch setting

* feat(Article): update OnDelete to author

* feat(Comment): update OnDelete to author

* feat(ArticleRepository): add existsByIdAndAuthor method

* feat(CommentRepository): add existsByIdAndAuthor method

* feat(NoPermissionException): add exception

* feat(NoPermissionCommentException): add exception

* feat(NoRowException): add exception

* feat(ArticleService): update Authentication and exception

* feat(CommentService): update Authentication and exception

* feat(UserService): update Authentication and exception

* feat(AuthAdvice): update exception handling

* feat(ArticleController): update Authentication

* feat(CommentController): update Authentication

* feat(UserController): update Authentication

* feat(application.properties): update DB setting

* test(ArticleServiceTest): test

* test(CommentServiceTest): test

* test(UserServiceTest): test crud and exist

* test(ControllerTest): add login method

* test(ArticleControllerTest): test to WebTestClient

* test(CommentControllerTest): test to WebTestClient

* test(UserControllerTest): test to WebTestClient

* test(HomeControllerTest): test to WebTestClient

hyperpace committed:
Revert "[댓글 기능] 에이든 미션 제출합니다. (#117)"

This reverts commit 920c9ead0ce9c0f1b75c2ab0d4beaee9b7ea1831.

Deocksoo committed:
Merge remote-tracking branch 'pair/hyperpace' into Deocksoo

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.properties
#	src/main/java/techcourse/myblog/MyBlogApplication.java
#	src/main/java/techcourse/myblog/domain/Article.java
#	src/main/java/techcourse/myblog/repository/ArticleRepository.java
#	src/main/java/techcourse/myblog/repository/UserRepository.java
#	src/main/java/techcourse/myblog/service/ArticleService.java
#	src/main/java/techcourse/myblog/service/UserService.java
#	src/main/java/techcourse/myblog/web/ArticleController.java
#	src/main/resources/application.properties
#	src/main/resources/static/js/index.js
#	src/main/resources/templates/article-edit.html
#	src/main/resources/templates/article.html
#	src/main/resources/templates/fragments/footer.html
#	src/main/resources/templates/fragments/header.html
#	src/main/resources/templates/index.html
#	src/main/resources/templates/login.html
#	src/main/resources/templates/mypage-edit.html
#	src/main/resources/templates/mypage.html
#	src/main/resources/templates/signup.html
#	src/main/resources/templates/user-list.html
#	src/test/java/techcourse/myblog/service/ArticleServiceTest.java
#	src/test/java/techcourse/myblog/service/UserServiceTest.java
#	src/test/resources/application.properties

Deocksoo committed:
resolve conflict error

Deocksoo committed:
set .gitignore file

Deocksoo committed:
Fix untraced files

Deocksoo committed:
refactor: commentDto의 이름을 RequestCommentDto로 변경합니다

Deocksoo committed:
refactor: Ajax로 데이터를 주고받도록 CommentController를 RestController로 변경

Deocksoo committed:
test: 테스트에 필요한 data.sql을 추가합니다

Deocksoo committed:
feat: logback.xml에 RollingFileAppener를 추가합니다

Deocksoo committed:
test: CommentControllerTest를 json 요청 테스트로 수정합니다

hyperpace committed:
feat: update application.properties

hyperpace committed:
feat: inital commit

hyperpace committed:
test(data.sql): add data.sql and test CommentController

hyperpace committed:
refactor: clean up code style

Commit:
fix: remove LOG_HOME in logback.xml

Commit:
Merge branch 'hyperpace' of https://github.com/hyperpace/jwp-blog into Deocksoo

Commit:
feat: 자동 배포를 위한 script 파일 추가

Deocksoo committed:
refactor: comment.js에서 불필요한 test 객체 제거

gracefulBrown reviewed (changes requested):
안녕하세요. 리뷰어 브라운 입니다.
ajax 기능 구현 및 테스트 구현 잘 해주셨습니다 👍 
테스트 중복제거가 인상깊네요~
몇가지 코멘트 남겼으니 확인해보시고 반영해보면 좋을 것 같아요!
궁금한 점 있으면 언제든 DM주세요 :)

Deocksoo committed:
refactor: UserControllerTest의 formatting을 수정합니다

Deocksoo committed:
refactor: logback.xml의 불필요한 주석을 제거하고 formatting 합니다

Deocksoo committed:
fix: article을 delete 하는 경우에 delete 요청이 두번 날아가는 버그를 수정합니다

Deocksoo committed:
feat: CommentController에 Comment 조회 기능을 추가합니다

조회 기능을 이용해 CommenControllerTest의 delete test에 삭제 후 조회하여 제대로 지워졌는지 확인하는 로직을
추가하였습니다.

Deocksoo committed:
fix: BlogControllerAdvice에서 CommentNotFoundException 처리 오류를 수정합니다

인자로 MisMatchAuthorException을 잘못 받고 있었기에 이를 수정합니다

Deocksoo commented:
언급해주신 피드백을 모두 반영해보는 것이 좋을것 같아 수정했습니다!
댓글 조회 기능이 지금 당장은 필요하지 않을 거라고 생각해서 만들지 않았었는데, 
나중에의 활용도 뿐만 아니라 controller를 테스트하는 것도 고려했을 때 만들어두는 것이 더 좋은것 같아요.

혹시 더 수정해야할 사항이 있으면 말씀해주세요 :)

gracefulBrown reviewed (approved):
피드백 반영 및 리팩토링 진행 잘 해주셨네요 👍 
몇가지 코멘트 남겼으니 확인해보시고
궁금한점은 언제든 DM 보내주세요!
수고 많으셨습니다!  머지할게요~
